### PR TITLE
[op](feat) commit dot_scaled fp8 case

### PR DESCRIPTION
**Pull request by liufengsheng-beep**
The ut test result is here:
root@triton:/home/l30058175/ta0718_main/test_0722# pytest -sv test_dot_scaled_fp8.py
====================================================================================================================== test session starts =======================================================================================================================
platform linux -- Python 3.11.13, pytest-9.0.2, pluggy-1.6.0 -- /usr/local/python3.11.13/bin/python3.11
cachedir: .pytest_cache
rootdir: /home/l30058175/ta0718_main/test_0722
plugins: anyio-4.10.0, rerunfailures-16.4, xdist-3.8.0, repeat-0.9.4
collected 16 items

test_dot_scaled_fp8.py::test_2D_scaled_dot_chain[1-7-64-fp8e4m3] [W722 07:04:29.825051308 NPUCachingAllocator.cpp:198] Warning: The current CANN and Soc Version require processing for 32 padding size, with memory allocation. (function operator())
PASSED
test_dot_scaled_fp8.py::test_2D_scaled_dot_chain[1-7-64-fp8e5m2] PASSED
test_dot_scaled_fp8.py::test_2D_scaled_dot_chain[15-128-64-fp8e4m3] PASSED
test_dot_scaled_fp8.py::test_2D_scaled_dot_chain[15-128-64-fp8e5m2] PASSED
test_dot_scaled_fp8.py::test_2D_scaled_dot_chain[128-64-64-fp8e4m3] PASSED
test_dot_scaled_fp8.py::test_2D_scaled_dot_chain[128-64-64-fp8e5m2] PASSED
test_dot_scaled_fp8.py::test_2D_scaled_dot_chain[189-175-64-fp8e4m3] PASSED
test_dot_scaled_fp8.py::test_2D_scaled_dot_chain[189-175-64-fp8e5m2] PASSED
test_dot_scaled_fp8.py::test_2D_scaled_dot_chain[1-1-128-fp8e4m3] PASSED
test_dot_scaled_fp8.py::test_2D_scaled_dot_chain[1-1-128-fp8e5m2] PASSED
test_dot_scaled_fp8.py::test_2D_scaled_dot_chain[35-37-128-fp8e4m3] PASSED
test_dot_scaled_fp8.py::test_2D_scaled_dot_chain[35-37-128-fp8e5m2] PASSED
test_dot_scaled_fp8.py::test_2D_scaled_dot_chain[62-64-192-fp8e4m3] PASSED
test_dot_scaled_fp8.py::test_2D_scaled_dot_chain[62-64-192-fp8e5m2] PASSED
test_dot_scaled_fp8.py::test_2D_scaled_dot_chain[31-27-256-fp8e4m3] PASSED
test_dot_scaled_fp8.py::test_2D_scaled_dot_chain[31-27-256-fp8e5m2] PASSED

====================================================================================================================== 16 passed in 20.98s =======================================================================================================================
root@triton:/home/l30058175/ta0718_main/test_0722#

# New contributor declaration
- [ ] I am not making a trivial change, such as fixing a typo in a comment.

- [ ] I have written a PR description following these
  [rules](https://cbea.ms/git-commit/#why-not-how).

- [ ] I have run `pre-commit run --from-ref origin/main --to-ref HEAD`.

- Select one of the following.
  - [ ] I have added tests.
    - `/test` for `lit` tests
    - `/unittest` for C++ tests
    - `/python/test` for end-to-end tests
  - [ ] This PR does not need a test because `FILL THIS IN`.

- Select one of the following.
  - [ ] I have not added any `lit` tests.
  - [ ] The `lit` tests I have added follow these [best practices](https://mlir.llvm.org/getting_started/TestingGuide/#filecheck-best-practices),
    including the "tests should be minimal" section. (Usually running Python code
    and using the instructions it generates is not minimal.)

### CODE REVIEW

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+    (31, 27, 256),
+    (31, 31, 256),
+    (32, 1, 256),
+    (32, 17, 256),
+    (32, 32, 256),
+]
+
+
+def generate_tensor_fp8(shape, dtype, seed=None, data_range=None,
+                        special_ratio=0.05,  # 插入 inf/-inf/nan的比例（仅当dtype为float类时生效）
```
Comment (hongziqi) [inline third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py]: It is recommended to use English comments.

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+    rhs_scale = tl.load(rhs_scale_ptr + Nidx[:, None] * (K // 32) + K1idx[None, :])
+
+    # First mmad: init=true equivalent (fresh accumulation).
+    first = tl.dot_scaled(lhs, lhs_scale, lhs_format, rhs, rhs_scale, rhs_format)
+    # Second mmad: init=false equivalent (accumulate into L0C).
+    result = tl.dot_scaled(lhs, lhs_scale, lhs_format, rhs, rhs_scale, rhs_format, acc=first)
+    tl.store(out_ptr + Midx[:, None] * N + Nidx[None, :], result)
+
+
+testlist = [
```
Comment (hongziqi) [inline third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py:81]: Why test so many cases? We recommend retaining typical shapes. UT should cover basic functions instead of pursuing generalization.

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+        if (extreme_ratio + precision_ratio + zero_ratio + special_ratio > 1):
+            raise ValueError(f"Total ratio over 100%, unable to allocate sampling:{extreme_ratio + precision_ratio}")
+
+        # 读字典取得精度小浮点数数据范围
+        precision_range = float_precision_dict[dtype]
+
+        # 生成原始一维全0数据，float32防溢出
+        data = np.zeros(total, dtype=np.float32)
+
+        # 计算各区间元素个数（其中极值区域为累计计数：按极小值附近，极小值本身，极大值附近，极大值本身顺序）
```
Comment (LH-123L) [inline third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py]: use english

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+    (64, 1, 192),
+    (64, 17, 192),
+    (64, 62, 192),
+    (64, 64, 192),
+    # K = 256 — small irregular
+    (1, 1, 256),
+    (1, 32, 256),
+    (3, 17, 256),
+    (5, 23, 256),
+    (7, 7, 256),
```
Comment (LH-123L) [inline third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py]: Delete some shapes to reduce access control time.

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py [hunk 5]
```diff
@@ -0,0 +1,387 @@
+import pytest
+import torch
```
Comment (LH-123L) [inline third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py:22]: Missing copyright information.

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+    (31, 9, 256),
+    (31, 27, 256),
+    (31, 31, 256),
+    (32, 1, 256),
+    (32, 17, 256),
+    (32, 32, 256),
+]
+
+
+def generate_tensor_fp8(shape, dtype, seed=None, data_range=None,
```
Comment (hongziqi) [inline third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py]: Should the `generate_tensor_fp8` functions be placed in the test_common script?

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py (hunk 1 above)
Comment (liufengsheng-beep) [reply third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py]: done

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py (hunk 2 above)
Comment (liufengsheng-beep) [reply third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py:81]: have delete many cases

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py (hunk 3 above)
Comment (liufengsheng-beep) [reply third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py]: done

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py (hunk 4 above)
Comment (liufengsheng-beep) [reply third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py]: done

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py (hunk 5 above)
Comment (liufengsheng-beep) [reply third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py:22]: done

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py (hunk 6 above)
Comment (liufengsheng-beep) [reply third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py]: move the func to test_common

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py (hunk 2 above)
Comment (hongziqi) [reply third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py:81]: There are still excessive test cases. Further reduction is recommended.

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+    # K = 256 — medium irregular / near UB limit
+    (25, 7, 256),
+    (31, 27, 256),
+]
+
+
+@pytest.mark.skipif(not is_compile_on_910_95, reason="only support in A5")
+@pytest.mark.parametrize("dtype", ["fp8e4m3", "fp8e5m2"])
+@pytest.mark.parametrize("M, N, K", testlist)
+def test_2D_scaled_dot_chain(M, N, K, dtype):
```
Comment (hongziqi) [inline third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py:96]: Please attach test results in the PR description. The current CI does not run pytest_ut on Ascend 950.

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+    # First mmad: init=true equivalent (fresh accumulation).
+    first = tl.dot_scaled(lhs, lhs_scale, lhs_format, rhs, rhs_scale, rhs_format)
+    # Second mmad: init=false equivalent (accumulate into L0C).
+    result = tl.dot_scaled(lhs, lhs_scale, lhs_format, rhs, rhs_scale, rhs_format, acc=first)
+    tl.store(out_ptr + Midx[:, None] * N + Nidx[None, :], result)
+
+
+testlist = [
+    # K = 64 — very small / edge
+    (1, 7, 64),
```
Comment (hongziqi) [inline third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py:82]: There are still excessive test cases. Further reduction is recommended.

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py (hunk 2 above)
Comment (liufengsheng-beep) [reply third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py:81]: > There are still excessive test cases. Further reduction is recommended.
done, only eight cases are left now

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py (hunk 8 above)
Comment (liufengsheng-beep) [reply third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py:82]: done, only eight cases are left now

File: third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py (hunk 7 above)
Comment (liufengsheng-beep) [reply third_party/ascend/unittest/pytest_ut/test_dot_scaled_fp8.py:96]: done, paste it in the description

**Review by hongziqi — APPROVED** [review]

**Review by WuTYSFG — APPROVED** [review]
